.github/workflows/default.yml

```diff
@@ -0,0 +1,49 @@
+name: default flow
+
+on:
+  push:
+    branches: [main, '[0-9].[0-9]']
+  pull_request:
+    branches-ignore: ['*docs', '*noci']
+  workflow_call:
+
+jobs:
+  build:
+    name: default build job
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
```

Comment:
pull python-version into an environment variable and use that.
Also, any reason to not just jump (or at least try) 3.11/latest?

Reply:
why not check on the most primitive one?

Reply:
Only because they're tools - for tooling I tend to try and use latest, but it's a preference, not a requirement.